main/src/ca/uwaterloo/flix/language/fmt/SimpleType.scala

```diff
@@ -343,10 +343,46 @@ object SimpleType {
   case class NonPredFieldType(name: String, tpe: SimpleType) extends PredicateFieldType
 
   /**
-    * Creates a simple type from the well-kinded type `t`.
+    * From a given `Type` return a `Symbol` if it does indeed have a symbol
+    *
+    * @param t The type
+    */
+  private def getSymbols(t : TypeConstructor): List[Symbol] = t match {
+    case TypeConstructor.Enum(sym, _) => List(sym)
+    case TypeConstructor.Struct(sym, _) => List(sym)
+    case TypeConstructor.RestrictableEnum(sym, _) => List(sym)
+    case _ => List()
+  }
+
+
+  /**
+    * Find the least amount of qualification needed to make a type unique from a list of types
     */
-  def fromWellKindedType(t0: Type): SimpleType = {
+  private def leastDistinct(t0: QualifiedSym with Locatable, wrt: List[Any] = List()): QualifiedSym = {
+    import ca.uwaterloo.flix.language.ast.{Name => AName}
+    wrt flatMap { x => x match {
+      case x0 : QualifiedSym => List(x0)
+      case _ => List()
+    }} map { t1 =>
+      if (t0.qname == t1.qname && t0.qnamespace != t1.qnamespace) {
+        //If the names are the same, but the ''true'' full names are also diffent
+        t0.qnamespace zip t1.qnamespace indexWhere { case (x0, x1) => x0 != x1 } match {
+          case -1 => t0 //If the namespaces are (somehow) the same, return the original type (this should not happen)
+          case i => //Otherwise, return a new type with the namespace up to the first difference
+            val newNamespace = t0.qnamespace.take(i + 1)
+            AName.mkQName(newNamespace, t0.qname,t0.loc)
```

Comment:
We try avoid this kind of string / list manipulation. We want to work a higher level.